File: frontend/src/app/controller/service/admin/commun/PaiementIrDetailAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {PaiementIrDetailDto} from 'src/app/controller/model/commun/PaiementIrDetail.model';
import {PaiementIrDetailCriteria} from 'src/app/controller/criteria/commun/PaiementIrDetailCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class PaiementIrDetailAdminService extends AbstractService<PaiementIrDetailDto, PaiementIrDetailCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/paiementIrDetail/');
    }

    public constrcutDto(): PaiementIrDetailDto {
        return new PaiementIrDetailDto();
    }

    public constrcutCriteria(): PaiementIrDetailCriteria {
        return new PaiementIrDetailCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/univ/service/facade/admin/commun/PaiementIrDetailAdminService.java
package ma.zs.univ.service.facade.admin.commun;

import java.util.List;
import ma.zs.univ.bean.core.commun.PaiementIrDetail;
import ma.zs.univ.dao.criteria.core.commun.PaiementIrDetailCriteria;
import ma.zs.univ.zynerator.service.IService;



public interface PaiementIrDetailAdminService extends  IService<PaiementIrDetail,PaiementIrDetailCriteria>  {

    List<PaiementIrDetail> findByPaiementIrId(Long id);
    int deleteByPaiementIrId(Long id);
    long countByPaiementIrId(Long id);
    List<PaiementIrDetail> findByTauxIrId(Long id);
    int deleteByTauxIrId(Long id);
    long countByTauxIrId(Long id);




}
